sql/insert.go

```diff
@@ -77,8 +77,17 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 
 	for rows.Next() {
 		values := rows.Values()
-		if len(values) != len(cols) {
-			return nil, fmt.Errorf("invalid values for columns: %d != %d", len(values), len(cols))
+		for i := 0; i < len(values); i++ {
+			if col := cols[i]; values[i] == parser.DNull && !col.Nullable {
+				return nil, fmt.Errorf("null value in column %q violates not-null constraint", col.Name)
+			}
+		}
+
+		for _, col := range cols[len(values):] {
+			if !col.Nullable {
+				return nil, fmt.Errorf("null value in column %q violates not-null constraint", col.Name)
+			}
+			values = append(values, parser.DNull)
```

Comment:
I believe you can do this:

```
  INSERT INTO kv3 VALUES (k) VALUES ('a')
```

We're only inserting into column `k` even though column `v` is NOT NULL.